src/producer.rs

```diff
@@ -1425,11 +1427,51 @@ impl Giganto {
                 match result {
                     Ok(record) if record != last_record => {
                         last_record = record.clone();
-                        time_serial += 1;
-                        if time_serial > 999 {
-                            time_serial = 1;
+
+                        // Extract timestamp from record and implement deduplication logic
+                        let current_timestamp = if let Some(utc_time) = record.get(3) {
+                            match crate::syslog::parse_sysmon_time(utc_time) {
+                                Ok(datetime) => {
+                                    if let Some(timestamp) = datetime.timestamp_nanos_opt() {
+                                        timestamp
+                                    } else {
+                                        failed_cnt += 1;
+                                        error!(
+                                            "failed to convert timestamp to nanos #{}",
+                                            next_pos.line()
+                                        );
+                                        continue;
+                                    }
+                                }
+                                Err(e) => {
+                                    failed_cnt += 1;
+                                    error!("failed to parse sysmon time #{}: {e}", next_pos.line());
+                                    continue;
+                                }
+                            }
+                        } else {
+                            failed_cnt += 1;
+                            error!("missing time field #{}", next_pos.line());
+                            continue;
+                        };
+
+                        // Implement timestamp deduplication logic
+                        if let Some(ref_ts) = reference_timestamp {
+                            if current_timestamp == ref_ts {
+                                // Same timestamp, increment offset
+                                timestamp_offset += 1;
+                            } else {
+                                // Different timestamp, update reference and reset offset
+                                reference_timestamp = Some(current_timestamp);
+                                timestamp_offset = 1;
+                            }
+                        } else {
+                            // First event, set reference timestamp
+                            reference_timestamp = Some(current_timestamp);
+                            timestamp_offset = 1;
```

Comment:
@octoaide
Since the first event can use the default timestamp_offset value of 0, there's no need to set it to 1. Could you please remove this line?